### ISSUE TITLE
SDL MacOS Mojave 10.4 black screen.

### ISSUE THREAD

**Issue by illume** [Homebrew/homebrew-core#33016]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Use SDL programs.
- What happened (include command output)
SDL programs show a black screen.
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

Use an SDL program, and it will show a black screen.
See upstream bug reports and discussion. 
https://bugzilla.libsdl.org/show_bug.cgi?id=4272
https://discourse.libsdl.org/t/macos-10-14-mojave-issues/25060

I'm not aware of a working patch at the time of writing.

**Comment by illume** [comment]
Note, this effects SDL1 and SDL2.

**Comment by illume** [comment]
Yes, I'm following the issue closely and will submit a PR when a fix is available.

ps. I have a question about process around here... I think perhaps I'm confused, as other "upstream issue" breakages are left open. I thought linking to the upstream bug would be useful for people tracking to see if this has been fixed yet, and that it would reduce other people having to submit the same issue again.  I know this is useful for people tracking the breakage, but if it's not something that this project wants then I won't link things up anymore. Is the difference "This appears to be a problem with the SDL projects, and not within Homebrew's configuration"? **So I only submit issues that are to do with breakages if the issue is not in upstream?**

pps. Probably a better word for SDL 1 is undead ;) There's an SDL 1 branch that collects fixes, and 'one' new release planned. There's more than 50k apps using SDL1... with users in the hundreds of millions(way more than homebrew).

**Comment by fxcoudert** [comment]
As you say, we have to walk a fine line: we can't handle bug reports for all packages we ship, but we want to make sure a. they do not come from Homebrew packaging, b. they are reported upstream.

Basically, once a bug is identified, analysed, and reported upstream, there is not much we can do until it's fixed. Keeping all issues open would be a nightmare for us in terms of prioritising support. People can still find the issue if they search for it!

**Comment by illume** [comment]
Makes sense. Thanks.

**Comment by Photosounder** [comment]
SDL 1.2 is actually still maintained (see https://hg.libsdl.org/SDL/shortlog/12354) but doesn't receive official releases anymore.

**Comment by Photosounder** [comment]
@illume SDL 2.0.9 has been released and fixes the problem apparently, but the formula for SDL2 still uses SDL 2.0.8 unfortunately.

**Comment by fxcoudert** [comment]
@Photosounder 2.0.9 has introduced another bug: see #33578

**Comment by Photosounder** [comment]
Oh, that's rather concerning, but what's the bug? All this says is "Created with brew bump-formula-pr."

**Comment by fxcoudert** [comment]
Sorry wrong PR: https://github.com/Homebrew/homebrew-core/pull/33781

**Comment by Photosounder** [comment]
Thank you, I just filed a bug for this on the SDL bug tracker: https://bugzilla.libsdl.org/show_bug.cgi?id=4385

**Comment by illume** [comment]
Here is the commit which fixes that issue: http://hg.libsdl.org/SDL/rev/d274fa9731b1
https://bugzilla.libsdl.org/show_bug.cgi?id=4367

I'm not sure when a new release is planned.

**Comment by Photosounder** [comment]
Yes sadly it doesn't seem like they intend on making a new release to fix this, so we might have to wait a few months until SDL 2.0.10.

**Comment by illume** [comment]
I guess we could make a PR which is 2.0.9 with that patch?
